src/Compilers/CSharp/Portable/SourceGeneration/CSharpGeneratorDriver.cs

```diff
@@ -17,9 +17,19 @@
 #nullable enable
 namespace Microsoft.CodeAnalysis.CSharp
 {
+    /// <summary>
+    /// A <see cref="GeneratorDriver"/> implementation for the CSharp language.
+    /// </summary>
     public sealed class CSharpGeneratorDriver : GeneratorDriver
     {
-        public CSharpGeneratorDriver(ParseOptions parseOptions, ImmutableArray<ISourceGenerator> generators, AnalyzerConfigOptionsProvider optionsProvider, ImmutableArray<AdditionalText> additionalTexts)
+        /// <summary>
+        /// Creates a new instance of <see cref="CSharpGeneratorDriver"/>
+        /// </summary>
+        /// <param name="parseOptions">The <see cref="CSharpParseOptions"/> that should be used when parsing generated files.</param>
+        /// <param name="generators">The generators that will run as part of this driver.</param>
+        /// <param name="optionsProvider">An <see cref="AnalyzerConfigOptionsProvider"/> that can be used to retrieve analyzer config values by the generators in this driver.</param>
+        /// <param name="additionalTexts">A list of <see cref="AdditionalText"/>s available to generators in this driver.</param>
+        public CSharpGeneratorDriver(CSharpParseOptions parseOptions, ImmutableArray<ISourceGenerator> generators, AnalyzerConfigOptionsProvider optionsProvider, ImmutableArray<AdditionalText> additionalTexts)
```

Comment:
Given the 2 new `Create` methods below, should these still be public?

Reply:
Currently working on a commit that does exactly that.